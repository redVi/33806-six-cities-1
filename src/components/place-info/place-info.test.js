import React from 'react';
import renderer from 'react-test-renderer';
import PlaceInfo from '@/components/place-info/place-info.jsx';

describe(`PlaceInfo`, () => {
  it(`renders correctly with props`, () => {
    const place = {
      title: `Beautiful & luxurious apartment at great location`,
      type: `Apartment`,
      price: 120,
      isFavorite: true,
      isPremium: true,
      rating: 4.1,
      previewImage: `apartment-01.jpg`,
    };
    const tree = renderer.create(<PlaceInfo {...place} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly with only required props`, () => {
    const place = {
      title: `Beautiful & luxurious apartment at great location`,
      price: 120,
      rating: 0,
    };
    const tree = renderer.create(<PlaceInfo {...place} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
