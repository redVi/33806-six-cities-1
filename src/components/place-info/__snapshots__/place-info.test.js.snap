// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceInfo renders correctly with only required props 1`] = `
<div
  className="place-card__info"
>
  <div
    className="place-card__price-wrapper"
  >
    <div
      className="place-card__price"
    >
      <b
        className="place-card__price-value"
      >
        €
        120
      </b>
      <span
        className="place-card__price-text"
      >
        / night
      </span>
    </div>
    <button
      className="place-card__bookmark-button  button"
      type="button"
    >
      <svg
        className="place-card__bookmark-icon"
        height="19"
        width="18"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>
  <div
    className="place-card__rating rating"
  >
    <div
      className="place-card__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
  </div>
  <h2
    className="place-card__name"
  >
    <a
      href="#"
    >
      Beautiful & luxurious apartment at great location
    </a>
  </h2>
  <p
    className="place-card__type"
  />
</div>
`;

exports[`PlaceInfo renders correctly with props 1`] = `
<div
  className="place-card__info"
>
  <div
    className="place-card__price-wrapper"
  >
    <div
      className="place-card__price"
    >
      <b
        className="place-card__price-value"
      >
        €
        120
      </b>
      <span
        className="place-card__price-text"
      >
        / night
      </span>
    </div>
    <button
      className="place-card__bookmark-button place-card__bookmark-button--active button"
      type="button"
    >
      <svg
        className="place-card__bookmark-icon"
        height="19"
        width="18"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>
  <div
    className="place-card__rating rating"
  >
    <div
      className="place-card__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "93%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
  </div>
  <h2
    className="place-card__name"
  >
    <a
      href="#"
    >
      Beautiful & luxurious apartment at great location
    </a>
  </h2>
  <p
    className="place-card__type"
  >
    Apartment
  </p>
</div>
`;
